lib/IRGen/GenConstant.cpp

```diff
@@ -0,0 +1,131 @@
+//===--- GenConstant.cpp - Swift IR Generation For Constant Values --------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+//  This file implements IR generation for constant values.
```

Comment:
Ok. I am fine with that. If we do that, we need to make sure that we add a verifier so we don't go astray again.